[GTK][WPE] Improve parsing of WEBKIT_DMABUF_RENDERER_BUFFER_FORMAT env var

GeorgesStavracas opened the pull request:
#### 6368ef08acb38d89c96530658fc01c911b6c1f25
<pre>
[GTK][WPE] Improve parsing of WEBKIT_DMABUF_RENDERER_BUFFER_FORMAT env var
<a href="https://bugs.webkit.org/show_bug.cgi?id=273209">https://bugs.webkit.org/show_bug.cgi?id=273209</a>

Reviewed by Carlos Garcia Campos.

Usually modifiers are shared, discussed, and referred to as hex numbers,
not boring base-10 numbers. Parse them as such. This allows us to take
modifiers from tools like &apos;wayland-info&apos; or GTK debug prints and stuff
it directly into the env var.

Also improve validation so that invalid modifiers aren&apos;t used.

Apply a similar change to WPE_DMABUF_BUFFER_FORMAT which follows the
same env var format.

* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreDMABuf.cpp:
(WebKit::AcceleratedBackingStoreDMABuf::preferredBufferFormats):
* Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
(wpe_view_get_preferred_dma_buf_formats):

Canonical link: <a href="https://commits.webkit.org/277978@main">https://commits.webkit.org/277978@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a5562fb5b7d58c142003b169d274d72144cb90f5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/49130 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28407 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/52138 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51928 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/45221 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/34380 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/25967 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/40153 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/50755 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/25968 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/42345 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21266 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/23425 "Passed tests") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/7454 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/45353 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/44026 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/53837 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/24208 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/20413 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/47472 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/25486 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/42550 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/46454 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10806 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/26295 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/25205 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos reviewed (approved):
This looks good to me, but please, do the same in `wpe_view_get_preferred_dma_buf_formats` in WPEView.cpp. WE have mostly the same code there.

GeorgesStavracas committed:
[GTK][WPE] Improve parsing of WEBKIT_DMABUF_RENDERER_BUFFER_FORMAT env var
https://bugs.webkit.org/show_bug.cgi?id=273209

Reviewed by Carlos Garcia Campos.

Usually modifiers are shared, discussed, and referred to as hex numbers,
not boring base-10 numbers. Parse them as such. This allows us to take
modifiers from tools like 'wayland-info' or GTK debug prints and stuff
it directly into the env var.

Also improve validation so that invalid modifiers aren't used.

Apply a similar change to WPE_DMABUF_BUFFER_FORMAT which follows the
same env var format.

* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreDMABuf.cpp:
(WebKit::AcceleratedBackingStoreDMABuf::preferredBufferFormats):
* Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
(wpe_view_get_preferred_dma_buf_formats):

Canonical link: https://commits.webkit.org/277978@main
